app/server/utils/ussd/kenya_ussd_state_machine.py

```diff
@@ -1,189 +1,548 @@
-      transitions from: :opt_out_of_market_place_pin_authorization, to: :exit_pin_blocked, guard: :blocked_pin
-    end
-
-    event :exchange_token do
-      transitions from: :exchange_token, to: :exchange_rate_pin_authorization, guard: :menu_one_selected
-      transitions from: :exchange_token, to: :exchange_token_agent_number_entry, guard: :menu_two_selected
-      transitions from: :exchange_token, to: :exit_invalid_menu_option
-    end
-
-    event :exchange_rate_pin_authorization do
-      transitions from: :exchange_rate_pin_authorization, to: :complete, guard: :authorize_pin, after: :fetch_user_exchange_rate
-      transitions from: :exchange_rate_pin_authorization, to: :exit_pin_blocked, guard: :blocked_pin
-    end
-
-    event :exchange_token_agent_number_entry do
-      transitions from: :exchange_token_agent_number_entry, to: :exchange_token_amount_entry, guard: :valid_token_agent, after: :save_exchange_agent_phone
-      transitions from: :exchange_token_agent_number_entry, to: :exit_invalid_token_agent
-    end
-
-    event :exchange_token_amount_entry do
-      transitions from: :exchange_token_amount_entry, to: :exchange_token_pin_authorization, guard: :valid_token_exchange_amount, after: :save_exchange_amount
-      transitions from: :exchange_token_amount_entry, to: :exit_invalid_exchange_amount
-    end
-
-    event :exchange_token_pin_authorization do
-      transitions from: :exchange_token_pin_authorization, to: :exchange_token_confirmation, guard: :authorize_pin
-      transitions from: :exchange_token_pin_authorization, to: :exit_pin_blocked, guard: :blocked_pin
-    end
-
-    event :exchange_token_confirmation do
-      transitions from: :exchange_token_confirmation, to: :complete, guard: :menu_one_selected, after: :process_exchange_token_request
-      transitions from: :exchange_token_confirmation, to: :exit, guard: :menu_two_selected
-      transitions from: :exchange_token_confirmation, to: :exit_invalid_menu_option
-    end
-
-    event :help do
-      transitions from: :help, to: :complete
-    end
-"""
\ No newline at end of file
+This file (kenya_ussd_state_machine.py) contains the KenyaUssdStateMachine responsible for managing the states of
+the ussd application as would apply to its use in kenya.
+It manages transitions between different states of the ussd system based on user input.
+The class contains methods responsible for validation of user input and processing user input to facilitate
+the services provided by the  ussd app.
+"""
+import re
+from transitions import Machine
+
+from server import db
+from server.models.user import User
+from server.models.ussd import UssdSession
+from server.utils.phone import proccess_phone_number, send_generic_message
+
+
+def get_user(query_filter):
+    user = User.query.filter_by(phone=proccess_phone_number(phone_number=query_filter, region="KE")).first()
+    if user is not None:
```

Comment:
what happens if user is None then?

Reply:
Should have had an exception there, I'll add one.